lib/mix/lib/mix.ex

```diff
@@ -675,47 +699,75 @@ defmodule Mix do
         Application.put_all_env(config, persistent: true)
         System.put_env(system_env)
 
-        installs_root =
-          System.get_env("MIX_INSTALL_DIR") || Path.join(Mix.Utils.mix_cache(), "installs")
-
-        version = "elixir-#{System.version()}-erts-#{:erlang.system_info(:version)}"
-        dir = Path.join([installs_root, version, id])
+        install_dir = opts[:__install_dir__] || install_dir(id)
 
         if opts[:verbose] do
-          Mix.shell().info("Mix.install/2 using #{dir}")
+          Mix.shell().info("Mix.install/2 using #{install_dir}")
         end
 
         if force? do
-          File.rm_rf!(dir)
+          File.rm_rf!(install_dir)
         end
 
+        lockfile = Path.join(install_dir, "mix.lock")
+
         config = [
           version: "0.1.0",
           build_embedded: false,
           build_per_environment: true,
           build_path: "_build",
-          lockfile: "mix.lock",
+          lockfile: lockfile,
```

Comment:
Do we need to change the lockfile? I think keep the previous configuration value... it doesn't need to be a full path.

Reply:
I added it to be more explicit but there is no need, will revert